### PR TITLE
cppcms 1.0.5

### PR DESCRIPTION
**Pull request by guoxiao**
Add cppcms formula

### CODE REVIEW

File: Library/Formula/cppcms.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    EOS
+    system ENV.cc, "-o", "hello", "-stdlib=libc++", "-lc++", "-lcppcms", "hello.cpp"
+    pid = fork { system "./hello", "-c", "config.json" }
+
+    sleep 1 # grace time for server start
+    begin
+      assert_match /Hello World/, shell_output("curl http://127.0.0.1:8080/hello")
+    ensure
+      Process.kill 9, pid
+      Process.kill 9, pid + 2 # kill child process
```
Comment (guoxiao) [inline Library/Formula/cppcms.rb]: A problem now: The test will fork a child process and is not easy to kill it. Maybe a better solution is needed.

**Comment by mistydemeo** [comment]
Can you add an openssl dependency?

**Comment by guoxiao** [comment]
Fails on ML. Do I need to add some special flags?

**Comment by guoxiao** [comment]
@BrewTestBot Test this please.

**Comment by DomT4** [comment]
The test-bot ignores pings from people who don't have admin access to it I'm afraid. I'll ask it for you.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
Looks like it hasn't compiled properly on 10.8 as the test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/30609/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/test_cppcms/

File: Library/Formula/cppcms.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  homepage "http://cppcms.com/wikipp/en/page/main"
+  url "https://downloads.sourceforge.net/project/cppcms/cppcms/1.0.5/cppcms-1.0.5.tar.bz2"
+  sha256 "84b685977bca97c3e997497f227bd5906adb80555066d811a7046b01c2f51865"
+
+  depends_on "cmake" => :build
+  depends_on "pcre"
+  depends_on "openssl"
+
+  def install
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", "."
```
Comment (mikemcquaid) [inline Library/Formula/cppcms.rb]: Use `std_cmake_args` instead.

File: Library/Formula/cppcms.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
+class Cppcms < Formula
+  desc "A Free High Performance Web Development Framework"
+  homepage "http://cppcms.com/wikipp/en/page/main"
+  url "https://downloads.sourceforge.net/project/cppcms/cppcms/1.0.5/cppcms-1.0.5.tar.bz2"
+  sha256 "84b685977bca97c3e997497f227bd5906adb80555066d811a7046b01c2f51865"
+
+  depends_on "cmake" => :build
+  depends_on "pcre"
+  depends_on "openssl"
+  depends_on :macos => :mavericks
```
Comment (dunn) [inline Library/Formula/cppcms.rb]: Is there no way to get it to compile on Mountain Lion?

Comment (guoxiao) [inline Library/Formula/cppcms.rb]: I tried several build options but couldn't get it work.

Comment (mikemcquaid) [inline Library/Formula/cppcms.rb]: Did you try a `fails_with` or `needs :cxx11`?

Comment (guoxiao) [inline Library/Formula/cppcms.rb]: In a previous PR #41848 @xu-cheng commented:
> ENV.cxx11 in the test won't work as the CFLAGS are never used,  Try to compile the test file in this way:
> ```
> system ENV.cc, "-std=c++11", "-stdlib=libc++", "-lc++", "-o", "test", "test.cpp"
> ```

It works well in `cereal`. However, in this formula I also added these params, but it doesn't work.

Comment (guoxiao) [inline Library/Formula/cppcms.rb]: > Did you try a fails_with or needs :cxx11?
Would you please tell me how to use these options in formulas? Or where is the documents/examples of these options? Thanks.

Comment (mikemcquaid) [inline Library/Formula/cppcms.rb]: I'm afraid they aren't documented but you can `grep` other formulae to see usage of `fails_with` or `needs :cxx11`

**Comment by DomT4** [comment]
Status update here?

**Comment by guoxiao** [comment]
`needs :cxx11` doesn't work here,  and I don't know how to use `fails_with` to handle this.

**Comment by DomT4** [comment]
How much does everyone care here about making it work on 10.8 and below?

**Comment by dunn** [comment]
@guoxiao did you try `needs :cxx11` above `def install`, and then `ENV.cxx11` inside the install block? Sometimes both are needed.

**Comment by mikemcquaid** [comment]
Thanks @guoxiao!

**Pull request closed without merging**